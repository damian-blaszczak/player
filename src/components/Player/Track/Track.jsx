import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import RedoIcon from '@material-ui/icons/Redo';
import UndoIcon from '@material-ui/icons/Undo';
import { ReactComponent as PlayIcon } from '../../../assets/icons/i_play.svg';
import { ReactComponent as PauseIcon } from '../../../assets/icons/i_pause.svg';
import secToTimeFormat from '../../../lib/utils/secToTimeFormat';

const TrackContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px 20px;
`;

const StyledImg = styled.img`
  max-width: 100%;
  min-width: 370px;
  min-height: 370px;
  max-height: 100%;
  border-radius: 5px;
`;

const StyledTrackName = styled.h3`
  font-size: 1.5rem;
  font-weight: 500;
`;

const MaterialIconContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const TrackNameSection = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin-top: 20px;
`;

const TrackControlsSection = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  margin-top: 20px;
`;

const StyledTimeAmount = styled.span`
  font-size: 1.4rem;
`;

const playPauseIconStyles = {
  width: '100px',
  height: '100px'
};

const fontSizeStyle = { fontSize: '4rem' };

const Track = ({
  trackImage,
  trackName,
  fastForward,
  rewind,
  playing,
  setPlaying,
  nextTrack,
  previousTrack,
  curTime,
  duration,
  preloadImages
}) => {

  useEffect(() => {
    preloadImages()
  }, [preloadImages]);

  return (
    <TrackContainer>
      <StyledImg src={trackImage} alt={`${trackName}-cover`} />
      <TrackNameSection>
        <ChevronLeftIcon data-testid="previousTrack" fontSize="large" onClick={previousTrack} />
        <StyledTrackName>{trackName}</StyledTrackName>
        <ChevronRightIcon data-testid="nextTrack" fontSize="large" onClick={nextTrack} />
      </TrackNameSection>
      <TrackControlsSection>
        <MaterialIconContainer>
          <UndoIcon style={fontSizeStyle} onClick={rewind} />
          <StyledTimeAmount>15</StyledTimeAmount>
        </MaterialIconContainer>
        {playing
          ? <PauseIcon style={playPauseIconStyles} onClick={() => setPlaying(false)} />
          : <PlayIcon style={playPauseIconStyles} onClick={() => setPlaying(true)} />
        }
        <MaterialIconContainer>
          <RedoIcon style={fontSizeStyle} onClick={fastForward} />
          <StyledTimeAmount>30</StyledTimeAmount>
        </MaterialIconContainer>
      </TrackControlsSection>
      <StyledTimeAmount>{secToTimeFormat(curTime)} / {secToTimeFormat(duration)}</StyledTimeAmount>
    </TrackContainer>
  )
}

Track.propTypes = {
  trackImage: PropTypes.string,
  trackName: PropTypes.string,
  fastForward: PropTypes.func,
  rewind: PropTypes.func,
  playing: PropTypes.bool,
  setPlaying: PropTypes.func,
  nextTrack: PropTypes.func,
  previousTrack: PropTypes.func,
  preloadImages: PropTypes.func
};

export default Track;
