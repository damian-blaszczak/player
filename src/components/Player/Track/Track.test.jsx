import React from 'react';
import { render } from '@testing-library/react';
import Track from './Track';

it('renders correctly', () => {
  const props = {
    trackImage: 'url',
    trackName: 'name',
    fastForward: jest.fn(),
    rewind: jest.fn(),
    playing: true,
    setPlaying: jest.fn(),
    nextTrack: jest.fn(),
    previousTrack: jest.fn(),
    preloadImages: jest.fn()
  };
  const { container } = render(
    <Track {...props} />
  );
  expect(container).toMatchSnapshot();
});
