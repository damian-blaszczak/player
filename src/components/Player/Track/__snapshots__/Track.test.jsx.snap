// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-AxjAm cAcpIu"
  >
    <img
      alt="name-cover"
      class="sc-AxirZ gWTLdP"
      src="url"
    />
    <div
      class="sc-AxhUy jpygBF"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
        data-testid="previousTrack"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
      <h3
        class="sc-AxiKw chNGWo"
      >
        name
      </h3>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
        data-testid="nextTrack"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </div>
    <div
      class="sc-AxgMl hIAvRk"
    >
      <div
        class="sc-AxhCb kakxRK"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="font-size: 4rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
          />
        </svg>
        <span
          class="sc-AxheI hrWohY"
        >
          15
        </span>
      </div>
      <svg
        style="width: 100px; height: 100px;"
      >
        i_pause.svg
      </svg>
      <div
        class="sc-AxhCb kakxRK"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="font-size: 4rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"
          />
        </svg>
        <span
          class="sc-AxheI hrWohY"
        >
          30
        </span>
      </div>
    </div>
    <span
      class="sc-AxheI hrWohY"
    >
      NaN:0NaN
       / 
      NaN:0NaN
    </span>
  </div>
</div>
`;
